base/range.jl

```diff
@@ -115,9 +124,47 @@ convert(::Type{T}, r::AbstractRange) where {T<:AbstractRange} = r isa T ? r : T(
 
 ## ordinal ranges
 
+"""
+    OrdinalRange{T, S} <: AbstractRange{T}
+
+Supertype for ordinal ranges with elements of type `T` with
+spacing(s) of type `S`.
+An ordinal range is one whose values have a well-defined ordering.
```

Comment:
This may be true, but I think it should be true of all ranges.  I think the key property that we're going after here — and the reasoning  behind the word `Ordinal` in this name — is that its subtypes have always-exact steps that are multiples of `oneunit`, and `T` should be a "discrete" datatype where it's not possible to represent values smaller than `oneunit`.

I think.  Can anyone back me up on this?